src/main/java/uk/gov/companieshouse/web/accounts/transformer/smallfull/impl/CreditorsAfterOneYearTransformerImpl.java

```diff
@@ -89,8 +109,85 @@ private void populatePreviousPeriodForWeb(CreditorsAfterOneYearApi creditorsAfte
         }
     }
 
-    @Override
-    public CreditorsAfterOneYearApi getCreditorsAfterOneYearApi(CreditorsAfterOneYear creditorsAfterOneYear) {
-        return null;
+    private void setCurrentPeriodOnApiModel(CreditorsAfterOneYear creditorsAfterOneYear,
+            CreditorsAfterOneYearApi creditorsAfterOneYearApi) {
+        CurrentPeriod currentPeriod = new CurrentPeriod();
+
+        if (creditorsAfterOneYear.getDetails() != null
```

Comment:
You could replace this check with 
```
if (StringUtils.isNotEmpty(creditorsAfterOneYear.getDetails()) {
      currentPeriod.setDetails(creditorsAfterOneYear.getDetails());
}
```

The `isNotEmpty` will catch `null`'s and empty strings (`""`). There's also the option of `isNotBlank` which will also catch whitespace (i.e.: `"       "`).

Reply:
So details was storing in Mongo as an empty string - and therefore always showing on the review screen. This method checks if details is not null, but empty - set to null, otherwise, set the details field. Changed to use isBlank rather than "".

```
 if (creditorsAfterOneYear.getDetails() != null
                && StringUtils.isBlank(creditorsAfterOneYear.getDetails())){
            currentPeriod.setDetails(null);
        } else {
            currentPeriod.setDetails(creditorsAfterOneYear.getDetails());
        }
```

Reply:
With the logic I suggested above, the value is only ever set on `currentPeriod` if `creditorsAfterOneYear.getDetails()` is not `null` or empty. Since the member variable `details` on `currentPeriod` is `null` by default, this would be sufficient.

At the moment, if `creditorsAfterOneYear.getDetails()` returns `null`, we explicitly call `currentPeriod.setDetails(null)`. It's just not as streamlined as it could be, though I won't block the PR for it